### PR TITLE
add likutim to metadata process

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: api/autoname_test.go [hunk 1]
```diff
@@ -337,3 +337,22 @@ func (suite *AutonameSuite) TestNameByTag() {
 	suite.Require().Nil(err)
 	suite.Len(names, 8, "len(names)")
 }
+
+/*
```
Comment (davgur) [inline api/autoname_test.go:341]: if need anything test here?

File: api/metadata_processor_test.go [hunk 2]
```diff
@@ -2268,6 +2299,23 @@ func (suite *MetadataProcessorSuite) assertContentUnit(metadata CITMetadata, ori
 		suite.False(missing, "Missing tag %s", x)
 	}
 
+	// likutim
+	likutim, err := models.ContentUnits(suite.tx,
```
Comment (davgur) [inline api/metadata_processor_test.go:2303]: here I get empty array and I dont see any unit on DB.
I was try add Units type LIKUTIM [here](https://github.com/Bnei-Baruch/mdb/blob/a07d223359b6b61a9f4d8f3e8703b8b23e5f3250/migrations/data/dev.sql) but its broke other tests

File: api/metadata_processor_test.go [hunk 3]
```diff
@@ -27,6 +28,7 @@ type MetadataProcessorSuite struct {
 }
 
 func (suite *MetadataProcessorSuite) SetupSuite() {
+	log.SetLevel(log.DebugLevel)
```
Comment (edoshor) [inline api/metadata_processor_test.go:31]: please don't commit this. Not everywhere we need debug level.

File: api/autoname_test.go (hunk 1 above)
Comment (edoshor) [reply api/autoname_test.go:341]: Sure you need to test. 
What if someone change something by mistake ? you go and manually fix the DB ?!
Tests ensure the code stays as intended.

File: api/metadata_processor_test.go [hunk 4]
```diff
@@ -2125,6 +2144,18 @@ func (suite *MetadataProcessorSuite) someTags() []string {
 	return uids
 }
 
+func (suite *MetadataProcessorSuite) someLikutim() []string {
```
Comment (edoshor) [inline api/metadata_processor_test.go:2147]: better get input with list of sources and create likutim units for them.
This way no need for `createLikutim(exec boil.Executor)`

File: api/metadata_processor_test.go [hunk 5]
```diff
@@ -2280,3 +2328,20 @@ func (suite *MetadataProcessorSuite) assertContentUnit(metadata CITMetadata, ori
 		suite.Empty(cu.R.ContentUnitsPersons, "Empty cu.R.ContentUnitsPersons")
 	}
 }
+
+func (suite *MetadataProcessorSuite) createLikutim(exec boil.Executor) {
```
Comment (edoshor) [inline api/metadata_processor_test.go:2332]: not needed

File: api/metadata_processor_test.go [hunk 6]
```diff
@@ -72,6 +75,7 @@ func (suite *MetadataProcessorSuite) TestDailyLesson() {
 		Part:           null.IntFrom(0),
 		Sources:        suite.someSources(),
 		Tags:           suite.someTags(),
+		Likutim:        suite.someLikutim(),
```
Comment (edoshor) [inline api/metadata_processor_test.go:78]: testing likutim in `TestDdailyLesson` is enough. No real need in each and every flow.

File: api/autoname_test.go (hunk 1 above)
Comment (davgur) [reply api/autoname_test.go:341]: added

File: api/metadata_processor_test.go (hunk 3 above)
Comment (davgur) [reply api/metadata_processor_test.go:31]: removed

File: api/metadata_processor_test.go (hunk 6 above)
Comment (davgur) [reply api/metadata_processor_test.go:78]: removed

File: api/metadata_processor_test.go (hunk 5 above)
Comment (davgur) [reply api/metadata_processor_test.go:2332]: removed

File: api/metadata_processor_test.go (hunk 4 above)
Comment (davgur) [reply api/metadata_processor_test.go:2147]: changed

**Comment by alexmbb** [comment]
@edoshor  can you review it?

File: api/autoname.go [hunk 7]
```diff
@@ -847,6 +857,29 @@ func nameByTagUID(exec boil.Executor, uid string, cNumber *int) (map[string]stri
 	return names, nil
 }
 
+func nameByLikutimUID(exec boil.Executor, uid string) (map[string]string, error) {
```
Comment (edoshor) [inline api/autoname.go:860]: Nothing here is special to type likutim. 
Maybe rename to something like `nameByUnit` so it sounds more generic

File: api/metadata_processor.go [hunk 8]
```diff
@@ -322,6 +322,51 @@ func doProcess(exec boil.Executor, metadata CITMetadata, original, proxy *models
 			return nil, errors.Wrap(err, "Associate tags")
 		}
 	}
+	if len(metadata.Likutim) > 0 {
+		log.Infof("Associating %d likutim", len(metadata.Likutim))
+		likutim, err := models.ContentUnits(exec,
+			qm.Select("distinct on (\"content_units\".id) \"content_units\".*"),
+			qm.WhereIn("uid in ?", utils.ConvertArgsString(metadata.Likutim)...)).
```
Comment (edoshor) [inline api/metadata_processor.go:329]: Maybe enforce type likutim here ?
Otherwise you might get UID of existing units but not of type likutim.

**Review by edoshor — APPROVED** [review]

**Pull request merged**